File: BOJ/nodejs/2547.js
/*
사탕 선생 고창영
boj.kr/2547
 */
const start = new Date().getTime();

const fs = require("fs");
const input = fs.readFileSync("input.txt").toString().split("\n");
// const input = fs.readFileSync("/dev/stdin").toString().split("\n");

let lc = 0;
const T = 1 * input[lc++];
const ans = new Array(T).fill("");

for (let t = 0; t < T; t++) {
  lc++;
  const N = 1 * input[lc++];

  let sum = 0n;
  for (let n = 0; n < N; n++) {
    let candy = BigInt(input[lc++]);
    sum += candy;
  }
  sum % BigInt(N) === 0n ? (ans[t] = "YES") : (ans[t] = "NO");
}

console.log(ans.join("\n"));

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);

File: BOJ/python3/2547.py
'''
사탕 선생 고창영
boj.kr/2547
'''
import sys

import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력


T = int(input())

ans = [""] * T;
for t in range(T):
    input()
    N = int(input())

    S = 0
    for _ in range(N):
        candy = int(input())
        S += candy
    
    if (S % N == 0): ans[t] = "YES"
    else: ans[t] = "NO"

for item in ans:
  print(item)

# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")
